backend/server.py

```
@@ -533,29 +776,83 @@ def clean_transcript(file_path):
-    if not subtitle_files:
-        return jsonify({"error": "No subtitles found"}), 404
+    # Use video_id-scoped file path to avoid race conditions between
+    # concurrent requests for different videos.
+    # Safety: subtitle_path is safe because VIDEO_ID_PATTERN (above) restricts
+    # video_id to [a-zA-Z0-9_-]{11}, preventing path traversal and injection.
+    subtitle_path = os.path.join("subtitles", f"{video_id}")
 
-    latest_subtitle = max(subtitle_files, key=os.path.getctime)
-    transcript_text = clean_transcript(latest_subtitle)
+    try:
+        subprocess.run(
```

Ruff (0.15.9)

[error] `subprocess` call: check for execution of untrusted input

(S603)